com.minres.coredsl/src/com/minres/coredsl/CoreDsl.xtext

```diff
@@ -162,34 +162,39 @@ fragment DeclarationSpecifier
 Attribute
     :  	DoubleLeftBracket type=ID ('=' params+=ConditionalExpression | '(' params+=ConditionalExpression (',' params+=ConditionalExpression)* ')')? DoubleRightBracket
     ;
-    
+
 TypeSpecifier
-	:	PrimitiveType 
-	|	CompositeType
-	|	EnumType
+	:	ValueTypeSpecifier
+	|	VoidTypeSpecifier
 	;
-    
-PrimitiveType
-	:	dataType+=DataTypes+ BitSizeSpecifier? 
+
+ValueTypeSpecifier
+	:	PrimitiveTypeSpecifier
+	|	CompositeTypeSpecifier
+	|	EnumTypeSpecifier
 	;
 
-fragment BitSizeSpecifier
-	:	'<' size+=PrimaryExpression (',' size+=PrimaryExpression ',' size+=PrimaryExpression ',' size+=PrimaryExpression)? '>'
+PrimitiveTypeSpecifier
+	:	IntegerTypeSpecifier
+	|	FloatTypeSpecifier
+	|	BoolTypeSpecifier
 	;
 
-EnumType
-    :   'enum' name=ID? '{' EnumeratorList ','? '}'
-    |   'enum' name=ID
-    ;
+IntegerTypeSpecifier
+	:	signedness=IntegerSignedness (shorthand=IntegerSizeShorthand | '<' size=PrimaryExpression '>')
+	|	shorthand=IntegerSizeShorthand
+	;
 
-fragment EnumeratorList:   enumerators+=Enumerator (',' enumerators+=Enumerator)*;
+FloatTypeSpecifier
+	:	shorthand=FloatSizeShorthand;
```

Comment:
I remember discussing this with @jopperm a while ago.
Our consensus was to drop it for now since we didn't see an immediate use case for custom float types.

Reply:
Ok.